diff --git a/app/crm/1stage.tsx b/app/crm/1stage.tsx
index c85f67f..797e3e7 100644
--- a/app/crm/1stage.tsx
+++ b/app/crm/1stage.tsx
@@ -13,7 +13,9 @@ import { useEffect, useState } from "react";
 import {
   ActivityIndicator,
   Alert,
+  KeyboardAvoidingView,
   Linking,
+  Platform,
   Modal,
   SafeAreaView,
   ScrollView,
@@ -270,7 +272,10 @@ export default function AllUsers() {
         transparent={true}
         onRequestClose={closeModal}
       >
-        <View style={styles.modalOverlay}>
+        <KeyboardAvoidingView
+          behavior={Platform.OS === "ios" ? "padding" : "height"}
+          style={styles.modalOverlay}
+        >
           <View style={styles.modalContainer}>
 
             {/* Modal Header */}
@@ -610,7 +615,7 @@ export default function AllUsers() {
 
             </ScrollView>
           </View>
-        </View>
+        </KeyboardAvoidingView>
       </Modal>
     </SafeAreaView>
   );
diff --git a/app/crm/2stage.tsx b/app/crm/2stage.tsx
index 36e4fe3..09c39d1 100644
--- a/app/crm/2stage.tsx
+++ b/app/crm/2stage.tsx
@@ -13,7 +13,9 @@ import { useEffect, useState } from "react";
 import {
   ActivityIndicator,
   Alert,
+  KeyboardAvoidingView,
   Linking,
+  Platform,
   Modal,
   SafeAreaView,
   ScrollView,
@@ -257,7 +259,10 @@ export default function StageTwo() {
         transparent={true}
         onRequestClose={closeModal}
       >
-        <View style={styles.modalOverlay}>
+        <KeyboardAvoidingView
+          behavior={Platform.OS === "ios" ? "padding" : "height"}
+          style={styles.modalOverlay}
+        >
           <View style={styles.modalContainer}>
 
             {/* Modal Header */}
@@ -523,7 +528,7 @@ export default function StageTwo() {
 
             </ScrollView>
           </View>
-        </View>
+        </KeyboardAvoidingView>
       </Modal>
     </SafeAreaView>
   );
diff --git a/app/crm/finalStage.tsx b/app/crm/finalStage.tsx
index 8aa0b8a..81eef5a 100644
--- a/app/crm/finalStage.tsx
+++ b/app/crm/finalStage.tsx
@@ -13,8 +13,10 @@ import { useEffect, useState } from "react";
 import {
     ActivityIndicator,
     Alert,
-    Linking,
-    Modal,
+    KeyboardAvoidingView,
+  Linking,
+    Platform,
+  Modal,
     SafeAreaView,
     ScrollView,
     StyleSheet,
@@ -195,7 +197,10 @@ export default function FinalStage() {
         transparent={true}
         onRequestClose={closeModal}
       >
-        <View style={styles.modalOverlay}>
+        <KeyboardAvoidingView
+          behavior={Platform.OS === "ios" ? "padding" : "height"}
+          style={styles.modalOverlay}
+        >
           <View style={styles.modalContainer}>
 
             {/* Modal Header */}
@@ -300,7 +305,7 @@ export default function FinalStage() {
 
             </ScrollView>
           </View>
-        </View>
+        </KeyboardAvoidingView>
       </Modal>
     </SafeAreaView>
   );
