diff --git a/app/crm/1stage.tsx b/app/crm/1stage.tsx
index f97e99d..ee8d1bd 100644
--- a/app/crm/1stage.tsx
+++ b/app/crm/1stage.tsx
@@ -476,7 +476,7 @@ export default function AllUsers() {
                     multiline
                     numberOfLines={4}
                     textAlignVertical="top"
-                    onFocus={() => setTimeout(() => scrollRef.current?.scrollToEnd({ animated: true }), 300)}
+                    onFocus={() => setTimeout(() => scrollRef.current?.scrollToEnd({ animated: true }), 500)}
                   />
 
                   <TouchableOpacity
diff --git a/app/crm/2stage.tsx b/app/crm/2stage.tsx
index 21f3cfe..98076cc 100644
--- a/app/crm/2stage.tsx
+++ b/app/crm/2stage.tsx
@@ -387,7 +387,7 @@ export default function StageTwo() {
                     multiline
                     numberOfLines={4}
                     textAlignVertical="top"
-                    onFocus={() => setTimeout(() => scrollRef.current?.scrollToEnd({ animated: true }), 300)}
+                    onFocus={() => setTimeout(() => scrollRef.current?.scrollToEnd({ animated: true }), 500)}
                   />
 
                   {/* Save Notes */}
